plugin.go

```diff
@@ -266,6 +267,20 @@ func (p *CompiledPlugin) Instance(ctx context.Context, config PluginInstanceConf
 	}
 	closers = append(closers, extism.Close)
 
+	// Instantiate all non-main modules first
+	instancedModules := make(map[string]api.Module)
+	for name, module := range p.modules {
+		instance, err := p.runtime.InstantiateModule(ctx, module, moduleConfig.WithName(name))
+		if err != nil {
+			for _, m := range instancedModules {
```

Comment:
What do you think about reusing `closers` here instead? If an error occurs, it will close all modules (`p.extism` too), also same thing can be added if error occurs when instantiating `main`, we will need to close other modules there too

Reply:
good idea!